llvm/test/Transforms/CorrelatedValuePropagation/min-max.ll

```diff
@@ -235,3 +231,33 @@ define i8 @test19(i8 %x) {
   %r = call i8 @llvm.smax(i8 %x, i8 42)
   ret i8 %r
 }
+
+declare void @body(i32)
+
+define void @test() {
+; CHECK-LABEL: @test(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    br label [[FOR_BODY:%.*]]
+; CHECK:       for.body:
+; CHECK-NEXT:    [[INDVAR:%.*]] = phi i32 [ 0, [[ENTRY:%.*]] ], [ [[INC:%.*]], [[FOR_BODY]] ]
+; CHECK-NEXT:    call void @body(i32 65535)
+; CHECK-NEXT:    [[INC]] = add nsw i32 [[INDVAR]], 1
+; CHECK-NEXT:    [[CMP:%.*]] = icmp slt i32 [[INDVAR]], 65535
+; CHECK-NEXT:    br i1 [[CMP]], label [[FOR_BODY]], label [[EXIT:%.*]]
+; CHECK:       exit:
+; CHECK-NEXT:    ret void
+;
+entry:
+  br label %for.body
+
+for.body:
+  %indvar = phi i32 [ 0, %entry ], [ %inc, %for.body ]
+  %smax = call i32 @llvm.smax.i32(i32 %indvar, i32 65535)
+  call void @body(i32 %smax)
+  %inc = add nsw i32 %indvar, 1
+  %cmp = icmp slt i32 %indvar, 65535
+  br i1 %cmp, label %for.body, label %exit
+
+exit:
+  ret void
+}
```

Comment:
I'm super confused about why this test case is affected by your change. This doesn't look like an at-use situation at all?

Reply:
Ah, I think this is actually not because of the at-use, but because you know check the non-strict predicate in both directions. This is also responsible for the changes above. I think those ones might not have used this fold at all previously and instead used the special return handling.

Reply:
So this is fine, but we should also have a test for the at-use case.

Reply:
>  check the non-strict predicate in both directions

You are right.

Reply:
> So this is fine, but we should also have a test for the at-use case.

https://alive2.llvm.org/ce/z/BBum_Z